src/client/datascience/jupyter/jupyterSession.ts

```diff
@@ -110,7 +110,9 @@ export class JupyterSession extends BaseJupyterSession {
             // Make sure it is idle before we return
             await this.waitForIdleOnSession(newSession, timeoutMS);
         } catch (exc) {
-            if (exc instanceof JupyterWaitForIdleError) {
+            // Don't swallow known exceptions.
+            if (exc instanceof BaseError) {
```

Comment:
Bug fix here, we were basically swallowing other errors that we are interested in.
Thanks to BaseError we now knoww what errors are thrown by us, those are not to be swallowed & ignored, instead we just re-throw them. Line 120, the original exception is lost.